### PR TITLE
JIRA: ABC-185 Bring back the stride array

### PR DESCRIPTION
**Pull request by clusty**
Change the semantic of curvePointCount back to a stride array (and rename accordingly) as per the demo team's request.

@judubu please also validate propter naming if have better ideas :)

### CODE REVIEW

File: com.unity.formats.alembic/Runtime/Scripts/Importer/AlembicCurves.cs [hunk 1]
```diff
@@ -17,9 +17,9 @@ public class AlembicCurves : MonoBehaviour
         /// </summary>
         public Vector3[] Positions => positionsList.GetArray();
         /// <summary>
-        /// Returns an array where: the i-th entry returns how many points are contained in the i-th curve.
+        /// Returns an array where: the i-th entry returns the starting index of the i-th curve ( the i-th curve is between [StrideArray[i]..StrideArray[i+1]) ).
```
Comment (judubu) [inline com.unity.formats.alembic/Runtime/Scripts/Importer/AlembicCurves.cs:20]: Shouldn't that be: `the i-th curve is between [StrideArray[i]..StrideArray[i+1]-1)`

Comment (judubu) [reply com.unity.formats.alembic/Runtime/Scripts/Importer/AlembicCurves.cs:20]: I'd say curveOffsets or curvePointIndices speaks to me more than strideArray

File: com.unity.formats.alembic/Runtime/Scripts/Importer/AlembicCurvesRenderer.cs [hunk 2]
```diff
@@ -143,27 +142,37 @@ void GeneratePlaneMesh(Mesh theMesh, IReadOnlyList<Vector3> positions, IReadOnly
...
             using (var particleUV = new NativeArray<Vector2>(positionsM.Length, Allocator.TempJob))
             using (var indices = new NativeArray<int>(indexArraySize, Allocator.TempJob))
             {
                 vertices.CopyFrom(positionsM);
-                curveCounts.CopyFrom(curvesCountsM);
-                var cnt = 0;
-                for (var i = 0; i < strideArray.Length; ++i)
+                strideArray.CopyFrom(curveOffsetM);
+
+                if (curveOffsetM.Length > 1)
```
Comment (judubu) [inline com.unity.formats.alembic/Runtime/Scripts/Importer/AlembicCurvesRenderer.cs:161]: Not sure I understand what's happening here with the offset length check.

File: com.unity.formats.alembic/Runtime/Scripts/Importer/AlembicCurvesRenderer.cs [hunk 3]
```diff
@@ -143,27 +142,37 @@ void GeneratePlaneMesh(Mesh theMesh, IReadOnlyList<Vector3> positions, IReadOnly
...
+                {
+                    var nativeArray = curveCounts;
```
Comment (judubu) [inline com.unity.formats.alembic/Runtime/Scripts/Importer/AlembicCurvesRenderer.cs:163]: Can you rename nativeArray to a more meaningful name?

File: com.unity.formats.alembic/Runtime/Scripts/Importer/AlembicCurvesRenderer.cs (hunk 2 above)
Comment (clusty) [reply com.unity.formats.alembic/Runtime/Scripts/Importer/AlembicCurvesRenderer.cs:161]: Will do comment

File: com.unity.formats.alembic/Runtime/Scripts/Importer/AlembicCurves.cs (hunk 1 above)
Comment (clusty) [reply com.unity.formats.alembic/Runtime/Scripts/Importer/AlembicCurves.cs:20]: `curveOffsets` it is...

File: com.unity.formats.alembic/Runtime/Unity.Formats.Alembic.Runtime.api [hunk 4]
```diff
@@ -29,8 +29,8 @@ namespace UnityEngine.Formats.Alembic.Importer
     [UnityEngine.DisallowMultipleComponent] public class AlembicCurves : UnityEngine.MonoBehaviour
     {
         public event UnityEngine.Formats.Alembic.Importer.AlembicCurves.OnUpdateDataHandler OnUpdateData;
-        public int[] CurvePointCount { get; }
         public UnityEngine.Vector3[] Positions { get; }
+        public int[] StrideArray { get; }
```
Comment (judubu) [inline com.unity.formats.alembic/Runtime/Unity.Formats.Alembic.Runtime.api:33]: That should be CurveOffsets right?

**Review by judubu — CHANGES_REQUESTED** [review]
I think there's one typo left. Beside that, great work.

File: com.unity.formats.alembic/Runtime/Unity.Formats.Alembic.Runtime.api (hunk 4 above)
Comment (clusty) [reply com.unity.formats.alembic/Runtime/Unity.Formats.Alembic.Runtime.api]: Auto-generated file. forgot to commit it :(
Should really port the unit test I did for recorder that checks for such things...

**Review by judubu — APPROVED** [review]
👍

**Review by cguer — CHANGES_REQUESTED** [review]
I get this error continuously when adding an `Alembic Curves Renderer` component on a GameObject that does not contain any curves:

```
IndexOutOfRangeException: Index 0 is out of range of '0' Length.
Unity.Collections.NativeArray`1[T].FailOutOfRangeError (System.Int32 index) (at <e414e10bfe5f45729ff122f3359de21b>:0)
Unity.Collections.NativeArray`1[T].CheckElementWriteAccess (System.Int32 index) (at <e414e10bfe5f45729ff122f3359de21b>:0)
Unity.Collections.NativeArray`1[T].set_Item (System.Int32 index, T value) (at <e414e10bfe5f45729ff122f3359de21b>:0)
Scripts.Importer.AlembicCurvesRenderer.GenerateLineMesh (UnityEngine.Mesh theMesh, UnityEngine.Vector3[] positionsM, System.Int32[] curveOffsetM) (at Library/PackageCache/com.unity.formats.alembic@85a6ef536ec6-1610650915220/Runtime/Scripts/Importer/AlembicCurvesRenderer.cs:171)
Scripts.Importer.AlembicCurvesRenderer.UpdateMesh (UnityEngine.Formats.Alembic.Importer.AlembicCurves curves) (at Library/PackageCache/com.unity.formats.alembic@85a6ef536ec6-1610650915220/Runtime/Scripts/Importer/AlembicCurvesRenderer.cs:50)
Scripts.Importer.AlembicCurvesRenderer.LateUpdate () (at Library/PackageCache/com.unity.formats.alembic@85a6ef536ec6-1610650915220/Runtime/Scripts/Importer/AlembicCurvesRenderer.cs:73)
```

Doing the same thing in the previous version of the package (2.2.0-epx.1) does not produce any errors.

I think it's better if there is no error but if this is on purpose we could make the message a little bit more explicit for users.

**Review by cguer — APPROVED** [review]
QA approved 👍!

Tested on:
- 2020.2.1f1 (Windows only)

**Pull request merged**